x/tokenregistry/keeper/keeper.go

```diff
@@ -84,63 +84,70 @@ func (k keeper) GetEntry(wl types.Registry, denom string) (*types.RegistryEntry,
 	return nil, errors.Wrap(errors.ErrKeyNotFound, "registry entry not found")
 }
 
-func (k keeper) SetToken(ctx sdk.Context, entry *types.RegistryEntry) {
-	wl := k.GetRegistry(ctx)
+func (k keeper) GetDenomPrefix(ctx sdk.Context, denom string) []byte {
+	return append(types.TokenDenomPrefix, []byte(denom)...)
+}
 
-	for i := range wl.Entries {
-		if wl.Entries[i] != nil && strings.EqualFold(wl.Entries[i].Denom, entry.Denom) {
-			wl.Entries[i] = entry
-			k.SetRegistry(ctx, wl)
-			return
-		}
-	}
-	wl.Entries = append(wl.Entries, entry)
-	k.SetRegistry(ctx, wl)
+func (k keeper) GetDenom(ctx sdk.Context, denom string) types.RegistryEntry {
+
+	var entry types.RegistryEntry
+	key := k.GetDenomPrefix(ctx, denom)
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(key)
+
+	k.cdc.MustUnmarshal(bz, &entry)
+
+	return entry
 }
 
-func (k keeper) RemoveToken(ctx sdk.Context, denom string) {
-	registry := k.GetRegistry(ctx)
-	updated := make([]*types.RegistryEntry, 0)
-	for _, t := range registry.Entries {
-		if t != nil && !strings.EqualFold(t.Denom, denom) {
-			updated = append(updated, t)
-		}
-	}
-	k.SetRegistry(ctx, types.Registry{
-		Entries: updated,
-	})
+// SetToken add a new denom
+func (k keeper) SetToken(ctx sdk.Context, entry *types.RegistryEntry) {
+	entry.Sanitize()
```

Comment:
Why does Sanitize() modify the struct that's passed in?  Seems like it should be modifying a copy

Reply:
That may be a bug, would make sense to modify a copy. That being said AddTokenMetadata calls this and it does not appear to cause any issues if its trimmed. msg_server Register calls this where it also should not be a problem.

Reply:
Ok, will use the copy instead of pointer.